crates/lib/src/qpu/api.rs

```diff
@@ -202,6 +205,59 @@ pub struct ExecutionOptions {
     #[doc = "The timeout to use for the request, defaults to 30 seconds. If set to `None`, then there is no timeout."]
     #[builder(default = "Some(Duration::from_secs(30))")]
     timeout: Option<Duration>,
+    #[doc = "If `true` and the user is authorized, jobs will bypass change protection against managed settings."]
+    #[builder(default = "None")]
+    api_options: Option<ApiExecutionOptions>,
+}
+
+/// Options avaialable when executing a job on a QPU, particular to the execution service's API.
+///
+/// Use [`Default`] to get a reasonable set of defaults, or start with [`ApiExecutionOptionsBuilder`]
+/// to build a custom set of options.
+#[derive(Builder, Clone, Debug, Default)]
+#[allow(clippy::module_name_repetitions)]
+pub struct ApiExecutionOptions {
+    /// the inner proto representation
+    inner: InnerApiExecutionOptions,
+}
+
+impl PartialEq for ApiExecutionOptions {
+    /// implement Eq
+    fn eq(&self, other: &Self) -> bool {
+        self.inner.bypass_settings_protection == other.inner.bypass_settings_protection
```

Comment:
This is needed because `InnerApiExecutionOptions` doesn't impl `PartialEq`? I worry about this manual implementation falling out of sync with the available options. It seems like it would be easy to update the API client without updating this function. Any way we can avoid that?

Reply:
It doesn't implement `Eq`, correct. I can update the implementation to use deconstruction so that the compiler will yell if more fields are added

```rs
    fn eq(&self, other: &Self) -> bool {
        let InnerApiExecutionOptions {
            bypass_settings_protection,
        } = self.inner;
        bypass_settings_protection == other.inner.bypass_settings_protection
    }
```

Reply:
Fixed with the same issue as https://github.com/rigetti/qcs-sdk-rust/pull/415#pullrequestreview-1809872395